CMakeLists.txt

```diff
@@ -50,8 +50,11 @@ endif()
 
 option(SPDLOG_BUILD_ALL "Build all artifacts" OFF)
 
-# build shared option
-option(SPDLOG_BUILD_SHARED "Build shared library" OFF)
+# Retain backward compatibility
+# TODO: Remove in 2.x
+if (DEFINED SPDLOG_BUILD_SHARED)
+    option(BUILD_SHARED_LIBS "Build shared library" ${SPDLOG_BUILD_SHARED})
+endif()
```

Comment:
No need to redefine BUILD_SHARED_LIBS. 

just change line line 127 to 
```if(SPDLOG_BUILD_SHARED OR BUILD_SHARED_LIBS)``` and we are done. Single line change is all this PR can reduced to.

Reply:
Well, the idea was to use only one `add_library()`, which internally checks for `BUILD_SHARED_LIBS`.